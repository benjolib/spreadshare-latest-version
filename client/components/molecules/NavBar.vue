<template>
  <div>
    <nav :class="$style.nav">
      <div :class="$style.navContent">
        <div>

          <app-logo :class="$style.marginLogo" />
          <nav-tab
            :class="$style.marginTabs"
            title="Feed"
            image="/header-feed-bird.svg"
            link="/feed"
            color="#ea612d"
          />
          <nav-tab
            :class="$style.marginTabs"
            title="Explore"
            image="/header-explore-whale.svg"
            link="/"
            color="#2c247f"
          />
          <nav-tab
            :class="$style.marginTabs"
            title="Collabs"
            image="/header-collabs-octopus.svg"
            link="/collabs"
            color="#2860f5"
          />
        </div>
        <div>
          <button-login />
        </div>
      </div>
    </nav>
  </div>
</template>

<script>
// @flow

import AppLogo from '@atoms/AppLogo'
import NavTab from '@atoms/NavTab'
import ButtonLogin from '@atoms/ButtonLogin'

export default {
  components: { AppLogo, NavTab, ButtonLogin },
}
</script>

<style lang="scss" module>
@import '@design';

.nav {
  position: fixed;
  top: 0; /* stylelint-disable-line length-zero-no-unit */
  left: 0; /* stylelint-disable-line length-zero-no-unit */
  z-index: 999;
  display: flex;
  justify-content: center;
  width: 100%;
  height: 78px;
  background: $color-navbar-bg;
  border-bottom: 1px solid rgb(228, 228, 228);

  .navContent {
    position: relative;
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    max-width: 1260px;
    padding: 0 12px;

    @include mq(large-and-xlarge-devices) {
      padding: 0 24px;
    }
  }
}

.marginLogo {
  margin-right: 36px;
}

.marginTabs {
  margin-right: 8px;
  @include mq(large-and-xlarge-devices) {
    margin-right: 24px;
  }
}
</style>
