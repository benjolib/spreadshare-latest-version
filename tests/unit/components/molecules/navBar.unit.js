// @flow

import NavBar from '@molecules/NavBar'
import { shallowMount } from '@vue/test-utils'

describe('navBar', () => {
  const wrapper = shallowMount(NavBar)

  describe('unit tests', () => {
    test('is a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(wrapper.is(NavBar)).toBeTruthy()
    })

    test('should have feed tab', () => {
      const tabs = wrapper.findAll('navtab-stub')
      const tab = tabs.at(0)
      expect(tab.attributes().color).toBe('#ea612d')
      expect(tab.attributes().link).toBe('/feed')
      expect(tab.attributes().title).toBe('Feed')
      expect(tab.attributes().image).toBe('/header-feed-bird.svg')
      expect(tab.classes()).toContain('marginTabs')
    })

    test('should have explore tab', () => {
      const tabs = wrapper.findAll('navtab-stub')
      const tab = tabs.at(1)
      expect(tab.attributes().color).toBe('#2c247f')
      expect(tab.attributes().link).toBe('/')
      expect(tab.attributes().title).toBe('Explore')
      expect(tab.attributes().image).toBe('/header-explore-whale.svg')
      expect(tab.classes()).toContain('marginTabs')
    })

    test('should have collabs tab', () => {
      const tabs = wrapper.findAll('navtab-stub')
      const tab = tabs.at(2)
      expect(tab.attributes().color).toBe('#2860f5')
      expect(tab.attributes().link).toBe('/collabs')
      expect(tab.attributes().title).toBe('Collabs')
      expect(tab.attributes().image).toBe('/header-collabs-octopus.svg')
      expect(tab.classes()).toContain('marginTabs')
    })

    test('should have app logo', () => {
      const logo = wrapper.find('applogo-stub')
      expect(logo.classes()).toContain('marginLogo')
    })

    test('should have button login', () => {
      expect(wrapper.find('buttonlogin-stub').exists()).toBeTruthy()
    })
  })

  describe('snapshot tests', () => {
    test('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
