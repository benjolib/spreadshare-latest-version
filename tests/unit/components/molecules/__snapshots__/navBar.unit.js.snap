// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navBar snapshot tests should match the snapshot 1`] = `
<div>
  <nav
    class="nav"
  >
    <div
      class="navContent"
    >
      <div>
        <applogo-stub
          class="marginLogo"
        />
         
        <navtab-stub
          class="marginTabs"
          color="#ea612d"
          image="/header-feed-bird.svg"
          link="/feed"
          title="Feed"
        />
         
        <navtab-stub
          class="marginTabs"
          color="#2c247f"
          image="/header-explore-whale.svg"
          link="/"
          title="Explore"
        />
         
        <navtab-stub
          class="marginTabs"
          color="#2860f5"
          image="/header-collabs-octopus.svg"
          link="/collabs"
          title="Collabs"
        />
      </div>
       
      <div>
        <buttonlogin-stub />
      </div>
    </div>
  </nav>
</div>
`;
